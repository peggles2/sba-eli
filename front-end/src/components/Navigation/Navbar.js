import React, { Component } from 'react';
import { Menu, Dropdown } from 'semantic-ui-react';

export default class Navbar extends Component {
  state = {}

  render() {

    const learningPathOptions = [
      {
        text: "LP 1",
        value: "LP 1"
      },
      {
        text: "LP 2",
        value: "LP 2"
      },
      {
        text: "LP 3",
        value: "LP 3"
      },
    ]

    return(
      <Menu>
        <Dropdown placeholder='Learning Paths' selection options={learningPathOptions} />
      </Menu>
    )
  }
};

