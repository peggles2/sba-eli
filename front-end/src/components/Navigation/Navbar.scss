@import "../../assets/styles/base";

.navbar.ui.menu {
  height: 65px;
  background-color: $dark-blue;

  a {
    padding-bottom: 0;
  }

  .dropdown {
    padding-bottom: 0;
    margin-right: 20px;
  }

  #navbar-learning-paths.active {
    background-color: $grey1;

    .learning-path-item {
      background-color: $grey1 !important;
    }

    .learning-path-item:hover {
      background-color: $light-blue !important;

      .text {
        color: white;
      }
    }
  }

  .dropdown.visible {
    .text {
      color: black;
    }

    img {
      margin-right: 20px;
    }

    .topic-summary {
      margin-left: 15px;

      .path-description {
        white-space: normal;
        margin-top: 15px;
      }
    }
  }


  .search-bar {
    padding: 0;
    padding-top: 13px;

    .fields {
      margin-bottom: 0;

      .search-input {
        padding-right: 0;
      }
    }

    .button {
      background-color: $light-blue;
      color: white;
    }
  }

  .text {
    color: white;
    font-size: 1.2em;
  }

  img.logo {
    width: 2.5em;
  }

  .registrationButtons {
    padding-bottom: 0;

    .button {
      background-color: $dark-blue;
      color: white;
    }
  }

  .userButtons {
    color: white;

    a {
      color: white;
    }

    padding-bottom: 0;
  }

  .greeting {
    padding-bottom: 11px;
  }

  .logoutLink {
    margin-right: 90px;
    cursor: pointer;
  }
}

.ui.button.invisible {
  display: none;
}