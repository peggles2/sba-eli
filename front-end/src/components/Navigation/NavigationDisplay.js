import React from 'react';
import {Route, Switch} from 'react-router-dom';
import Dashboard from '../Dashboard/Dashboard';
import Error from "../Error";
import LearningEvent from "../LearningEvent/LearningEvent";
import LearningEvents from "../LearningEvent/LearningEvents";
import LearningObjective from "../LearningObjective/LearningObjective";
import LearningObjectives from "../LearningObjective/LearningObjectives";
import LearningPath from "../LearningPath/LearningPath";
import LearningPaths from "../LearningPath/LearningPaths";
import SignUpForm from "../SignUpForm/SignUpForm";
import SearchPage from "../Search/SearchPage";

const NavigationDisplay = () => {
  return (
      <Switch>
        <Route
            exact
            path="/"
            component={Dashboard} />
        <Route
            exact
            path="/signup"
            component={SignUpForm} />
        <Route
            exact
            path="/learning_paths"
            component={LearningPaths} />
        <Route
            exact
            path="/learning_paths/:id"
            component={LearningPath} />
        <Route
            exact
            path="/learning_paths/:course_id/learning_objectives/"
            component={LearningObjectives}
        />
        <Route
            exact
            path="/learning_paths/:course_id/learning_objectives/:id"
            component={LearningObjective}
        />
        <Route
            exact
            path="/learning_paths/:course_id/learning_objectives/:module_id/learning_events/:id"
            component={LearningEvent}
        />
        <Route
            exact
            path="/learning_paths/:course_id/learning_objectives/:module_id/learning_events"
            component={LearningEvents}
        />
        <Route
            exact
            path="/search"
            component={SearchPage}
        />
        <Route component={Error} />
      </Switch>
  );
}

export default NavigationDisplay;
