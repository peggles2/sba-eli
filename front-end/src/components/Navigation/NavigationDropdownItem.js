import React from "react";
import {Dropdown} from 'semantic-ui-react'
import {Link} from "react-router-dom";

export default class NavigationDropdownItem extends React.Component {
  render() {
    return <Dropdown item text={this.props.name}>
      <Dropdown.Menu>
        <Dropdown.Item as={Link} to={this.props.path} text={this.props.name} />
      </Dropdown.Menu>
    </Dropdown>
  }
}