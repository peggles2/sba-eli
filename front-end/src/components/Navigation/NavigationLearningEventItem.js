import React, { Component } from "react";
import { Dropdown } from "semantic-ui-react";

export default class NavigationLearningEventItem extends Component {
  render() {
    return (
      <Dropdown.Item href={this.props.path} text={this.props.name} />
    );
  }
}
