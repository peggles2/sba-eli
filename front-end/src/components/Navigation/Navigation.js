import React from "react";
import "../../App.css";
import { Grid } from "semantic-ui-react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Dashboard from "../Dashboard/Dashboard";
import LearningPath from "../LearningPath/LearningPath";
import LearningPaths from "../LearningPath/LearningPaths";
import LearningObjective from "../LearningObjective/LearningObjective";
import LearningObjectives from "../LearningObjective/LearningObjectives";
import Navbar from "./Navbar";
import LearningEvent from "../LearningEvent/LearningEvent";
import LearningEvents from "../LearningEvent/LearningEvents";
import SearchPage from "../Search/SearchPage";
import SignUpForm from "../SignUpForm/SignUpForm";
import Error from "../Error";

const Navigation = () => {
  return (
    <div>
      <Router>
        <div>
          <Navbar />
          <Grid>
            <Grid.Row>
              <Grid.Column mobile={16} tablet={8} computer={16}>
                <NavigationDisplay />
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </div>
      </Router>
    </div>
  );
};

const NavigationDisplay = () => {
  return (
    <Switch>
      <Route exact path="/" component={Dashboard} />
      <Route exact path="/signup" component={SignUpForm} />
      <Route exact path="/learning_paths" component={LearningPaths} />
      <Route exact path="/learning_paths/:id" component={LearningPath} />
      <Route
        exact
        path="/learning_paths/:course_id/learning_objectives/"
        component={LearningObjectives}
      />
      <Route
        exact
        path="/learning_paths/:course_id/learning_objectives/:id"
        component={LearningObjective}
      />
      <Route
        exact
        path="/learning_paths/:id/learning_objectives/:topicId/learning_events/:eventId"
        component={LearningPath}
      />
      <Route
        exact
        path="/learning_paths/:course_id/learning_objectives/:module_id/learning_events"
        component={LearningEvents}
      />
      <Route exact path="/search" component={SearchPage} />

      <Route component={Error} />
    </Switch>
  );
};

export default Navigation;
