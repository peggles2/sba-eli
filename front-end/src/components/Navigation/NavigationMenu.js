import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import {Menu} from 'semantic-ui-react'
import NavigationDropdownItem from "./NavigationDropdownItem";
import axios from "axios";
import {Dropdown} from 'semantic-ui-react'

export default class NavigationMenu extends Component {
  state = {
    activeItem: 'dashboard',
    learningPaths: [],
    learningObjectives: [],
    learningEvents: []
  }

  componentDidMount() {
    this.fetchNavigationData();
  }

  fetchNavigationData() {
    const lpUrl = process.env.REACT_APP_SERVICE_HOST +
        "/learning_paths"

    axios.get(lpUrl)
        .then(res => res.data)
        .then(learningPathData => {
          const learningPaths = learningPathData;
          let learningObjectivesData = []
          this.setState({learningPaths})

          learningPaths.map(lp => {

                const loUrl = process.env.REACT_APP_SERVICE_HOST +
                    "/learning_objectives/" +
                    "?course_id=" +
                    lp.id

                axios.get(loUrl)
                    .then(res2 => {
                      learningObjectivesData[lp.id] = res2.data
                    })
                    .then(tmp => {
                      const learningObjectives = learningObjectivesData;
                      this.setState({learningObjectives})
                    })
                    .catch(error => {
                      console.error(error)
                    })
              }
          )
        })
        .catch(error => {
          console.error(error)
        })

  }

  populateLearningPaths() {
    let lp_nav = ''
    if (this.state.learningPaths.length > 0) {
      lp_nav =
          this.state.learningPaths.map(lp =>
              <div>
              <Menu.Item key={lp.id}
                         as={Link}
                         to={`/learning_paths/${lp.id}`}
                  // name={`learning_paths_${lp.id}`}
                  // active={activeItem === `learning_paths_${lp.id}`}
                  // onClick={this.handleItemClick}
                         text={lp.name}>
                {lp.name}
              </Menu.Item>
              {this.populateLearningObjectives(lp.id)}
              </div>
          )
    }
    return lp_nav
  }

  populateLearningObjectives(courseId) {
    console.log("course id " + courseId)
    console.log(this.state.learningObjectives[courseId])
    let lo_nav = ''
    if (this.state.learningObjectives[courseId] && this.state.learningObjectives[courseId].length > 0) {
      lo_nav =
          this.state.learningObjectives[courseId].map(lo =>
              <NavigationDropdownItem key={lo.id} path={`/learning_objectives/${lo.id}`} name={lo.name} docs={this.state.learningEvents} />
          )
    }
    return lo_nav
  }

  handleItemClick = (e, {name}) => this.setState({
    activeItem: name
  })

  render() {

    const {activeItem} = this.state

    return (
        <Menu vertical>

          <Menu.Item as={Link}
                     to='/'
                     name='dashboard'
                     active={activeItem === 'dashboard'}
                     onClick={this.handleItemClick}>
            Dashboard
          </Menu.Item>

          <Menu.Item as={Link}
                     to='/learning_paths'
                     name='learning_paths'
                     active={activeItem === 'learning_paths'}
                     onClick={this.handleItemClick}>
            Learning Paths
          </Menu.Item>
          <Menu.Menu>
            {this.populateLearningPaths()}
          </Menu.Menu>
        </Menu>
    )
  }
}