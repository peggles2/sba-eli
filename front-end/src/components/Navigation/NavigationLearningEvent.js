import React, { Component } from "react";
import axios from "axios";
import { Dropdown } from "semantic-ui-react";
import NavigationLearningEventItem from "./NavigationLearningEventItem";

export default class NavigationLearningEvent extends Component {
  constructor(props) {
    super(props);

    this.state = {
      learningEvents: []
    };
  }

  componentDidMount() {
    this.fetchData();
  }

  fetchData() {
    const url = process.env.REACT_APP_SERVICE_HOST + `/learning_events/`;

    const eventParams = {
      course_id: this.props.learningPathId,
      module_id: this.props.learningObjectiveId
    };

    axios
      .get(url, { params: eventParams })
      .then(res => {
        const learningEvents = res.data;
        this.setState({ learningEvents });
      })
      .catch(error => {
        console.log(error);
      });
  }

  renderLearningEvents(learningEvents = []) {
    if (learningEvents.length) {
      const course_id = this.props.learningPathId;
      const module_id = this.props.learningObjectiveId;
      return learningEvents.map(event => {
        const path =
          `/learning_paths/${course_id}` +
          `/learning_objectives/${module_id}` +
          `/learning_events/${event.id}`;
        return (
          <NavigationLearningEventItem
            key={event.id + course_id + module_id}
            path={path}
            name={event.title}
          />
        );
      });
    }

    return <NavigationLearningEventItem path="/" name="No data" />;
  }

  render() {
    return (
      <Dropdown item text={this.props.name}>
        <Dropdown.Menu>
          {this.renderLearningEvents(this.state.learningEvents)}
        </Dropdown.Menu>
      </Dropdown>
    );
  }
}
