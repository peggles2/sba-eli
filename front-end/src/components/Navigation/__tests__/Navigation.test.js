import React from 'react';
import ReactDOM from 'react-dom';
import Navigation from '../Navigation';
import NavigationMenu from '../NavigationMenu';
import { shallow, mount } from 'enzyme';

describe('Navigation', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Navigation/>);
    
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render a <div>', () => {
    const wrapper = shallow(<Navigation/>);

    expect(wrapper.find('div').length).toEqual(2);
  });

  it('should render a <NavigationMenu>', () => {
    const wrapper = shallow(<Navigation/>);

    expect(wrapper.find(NavigationMenu).exists()).toBe(true);
  });
});
