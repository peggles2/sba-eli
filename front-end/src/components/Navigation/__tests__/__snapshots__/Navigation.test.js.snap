// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar />,
        <BrowserRouter>
          <div>
            <Grid>
              <GridRow>
                <GridColumn
                  computer={3}
                  mobile={16}
                  tablet={8}
                >
                  <NavigationMenu />
                </GridColumn>
                <GridColumn
                  computer={13}
                  mobile={16}
                  tablet={8}
                >
                  <NavigationDisplay />
                </GridColumn>
              </GridRow>
            </Grid>
          </div>
        </BrowserRouter>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <Grid>
              <GridRow>
                <GridColumn
                  computer={3}
                  mobile={16}
                  tablet={8}
                >
                  <NavigationMenu />
                </GridColumn>
                <GridColumn
                  computer={13}
                  mobile={16}
                  tablet={8}
                >
                  <NavigationDisplay />
                </GridColumn>
              </GridRow>
            </Grid>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Grid>
              <GridRow>
                <GridColumn
                  computer={3}
                  mobile={16}
                  tablet={8}
                >
                  <NavigationMenu />
                </GridColumn>
                <GridColumn
                  computer={13}
                  mobile={16}
                  tablet={8}
                >
                  <NavigationDisplay />
                </GridColumn>
              </GridRow>
            </Grid>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <GridRow>
                <GridColumn
                  computer={3}
                  mobile={16}
                  tablet={8}
                >
                  <NavigationMenu />
                </GridColumn>
                <GridColumn
                  computer={13}
                  mobile={16}
                  tablet={8}
                >
                  <NavigationDisplay />
                </GridColumn>
              </GridRow>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <GridColumn
                    computer={3}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationMenu />
                  </GridColumn>,
                  <GridColumn
                    computer={13}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationDisplay />
                  </GridColumn>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavigationMenu />,
                    "computer": 3,
                    "mobile": 16,
                    "tablet": 8,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavigationDisplay />,
                    "computer": 13,
                    "mobile": 16,
                    "tablet": 8,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar />,
          <BrowserRouter>
            <div>
              <Grid>
                <GridRow>
                  <GridColumn
                    computer={3}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationMenu />
                  </GridColumn>
                  <GridColumn
                    computer={13}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationDisplay />
                  </GridColumn>
                </GridRow>
              </Grid>
            </div>
          </BrowserRouter>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              <Grid>
                <GridRow>
                  <GridColumn
                    computer={3}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationMenu />
                  </GridColumn>
                  <GridColumn
                    computer={13}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationDisplay />
                  </GridColumn>
                </GridRow>
              </Grid>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Grid>
                <GridRow>
                  <GridColumn
                    computer={3}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationMenu />
                  </GridColumn>
                  <GridColumn
                    computer={13}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationDisplay />
                  </GridColumn>
                </GridRow>
              </Grid>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <GridRow>
                  <GridColumn
                    computer={3}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationMenu />
                  </GridColumn>
                  <GridColumn
                    computer={13}
                    mobile={16}
                    tablet={8}
                  >
                    <NavigationDisplay />
                  </GridColumn>
                </GridRow>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <GridColumn
                      computer={3}
                      mobile={16}
                      tablet={8}
                    >
                      <NavigationMenu />
                    </GridColumn>,
                    <GridColumn
                      computer={13}
                      mobile={16}
                      tablet={8}
                    >
                      <NavigationDisplay />
                    </GridColumn>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <NavigationMenu />,
                      "computer": 3,
                      "mobile": 16,
                      "tablet": 8,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <NavigationDisplay />,
                      "computer": 13,
                      "mobile": 16,
                      "tablet": 8,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
